app/src/main/java/com/github/se/eduverse/viewmodel/ProfileViewModel.kt

```diff
@@ -26,6 +26,8 @@ open class ProfileViewModel(private val repository: ProfileRepository) : ViewMod
   open val searchState: StateFlow<SearchProfileState> = _searchState.asStateFlow()
   private val _usernameState = MutableStateFlow<UsernameUpdateState>(UsernameUpdateState.Idle)
   open val usernameState: StateFlow<UsernameUpdateState> = _usernameState.asStateFlow()
+  private val _followActionState = MutableStateFlow<FollowActionState>(FollowActionState.Idle)
+  open val followActionState: StateFlow<FollowActionState> = _followActionState.asStateFlow()
 
```

Comment:
StateFlow handling for tracking the status of the follow-up action is well implemented. However, make sure that the Loading state is clearly visible to the user, such as with a progress indicator or other visual cue.